### PR TITLE
Add unchecked variants to CachedRegex and CachedRegexBuilder + general housekeeping

### PR DESCRIPTION
**Pull request by yannleretaille**
@meh would be great if you could review this!

### CODE REVIEW

File: src/cache.rs [hunk 1]
```diff
@@ -147,6 +147,15 @@ impl CachedRegex {
 		Ok(CachedRegex::from(CachedRegexBuilder::new(cache, source)))
```
Comment (meh) [inline src/cache.rs]: Could you use `new_unchecked` here?

**Review by meh — CHANGES_REQUESTED** [review]
Looks good otherwise.

Can you also use the same commit format as the rest please? (semantic commit messages)

File: src/cache.rs (hunk 1 above)
Comment (yannleretaille) [reply src/cache.rs]: why? the syntax check for `CachedRegexBuilder` only happens in ` CachedRegexBuilder::build`, not in `CachedRegexBuilder::new`

**Comment by yannleretaille** [comment]
now using semantic commit messages!

**Comment by meh** [comment]
Thank you for your valuable contribution!

**Pull request merged**